File: Day_1/main.cpp
#include <algorithm>
#include <iostream>
#include <fstream>
#include <vector>
#include <map>
#include <string>

int main()
{
    std::fstream file;
    file.open("data.txt", std::fstream::in);
    
    std::string line = "";
    std::vector<int> column_1;
    std::vector<int> column_2;
    std::map<int, int> repetitions; // Other half of exercise

    while(std::getline(file, line)){
        int num_1;
        int num_2;
        
        size_t space = line.find(" ");
        column_1.emplace_back(std::stoi(line.substr(0, space)));
        column_2.emplace_back(std::stoi(line.substr(space)));
        if(repetitions.find(column_2.back()) == repetitions.end()){
            repetitions.emplace(std::make_pair(column_2.back(), 1));
        } else {
            repetitions[column_2.back()]++;
        }
    }

    std::sort(column_1.begin(), column_1.end());
    std::sort(column_2.begin(), column_2.end());
    
    int total = 0;
    int total_similarity = 0; // Other half of exercise
    for(int i = 0; i < column_1.size(); i++){
        total += std::abs(column_2[i] - column_1[i]);
        if(repetitions.find(column_1[i]) != repetitions.end()){
            total_similarity += column_1[i] * repetitions[column_1[i]];
        }
        
    }
    
    std::cout << "The distance between the two lists is: " << total << std::endl;
    std::cout << "The similarity of the two lists is: " << total_similarity << std::endl;
    return 0;
}

File: Day_1/main.py
column_1 = []
column_2 = []
repetitions = {} #Second half of exercise

with open('data.txt') as file:
    for line in file:
        space = line.find(" ")
        column_1.append(int(line[:space]))
        column_2.append(int(line[space:]))
        if(repetitions.get(column_2[-1]) == None):
            repetitions[column_2[-1]] = 1
        else:
            repetitions[column_2[-1]] += 1


column_1.sort()
column_2.sort()

total = 0
total_similarity = 0 #Second half of exercise
for i in range(0, len(column_1)):
    total += abs(column_2[i] - column_1[i])
    if(repetitions.get(column_1[i]) != None):
        total_similarity += column_1[i] * repetitions.get(column_1[i])

print("The distance between the two lists is: " + str(total))
print("The similarity between the two lists is: " + str(total_similarity))